src/com/dotmarketing/util/UtilMethods.java

```diff
@@ -1211,21 +1216,42 @@ public static String cleanFileSystemPathURI(String path) {
 
     public static String validateFileName(String fileName) throws IllegalArgumentException{
 
-        if (!isSet(fileName)  ||
-        		fileName.indexOf("..") != -1 || fileName.indexOf("WEB-INF") != -1 || fileName.indexOf("META-INF") != -1 || fileName.indexOf("!") != -1
-                || fileName.indexOf(":") != -1 || fileName.indexOf(";") != -1 || fileName.indexOf(";") != -1 || fileName.indexOf("&") != -1
-                || fileName.indexOf("?") != -1 || fileName.indexOf("$") != -1 || fileName.indexOf("*") != -1 || fileName.indexOf("\"") != -1
-                || fileName.indexOf("/") != -1 || fileName.indexOf("[") != -1 || fileName.indexOf("]") != -1 || fileName.indexOf("=") != -1
-                || fileName.indexOf("|") != -1 || fileName.indexOf(",") != -1 || fileName.indexOf("#") != -1 || fileName.indexOf("{") != -1
-                || fileName.indexOf("}") != -1 || fileName.indexOf("\\") != -1) {
-            throw new IllegalArgumentException("Invalid Filename passed in: " + fileName);
+        if (!isSet(fileName) || REGEX_FILENAME_INVALID_CHARS.matcher(fileName).matches()) {
+
+        	throw new IllegalArgumentException("Invalid Filename passed in: " + fileName);
 
         } else {
             return fileName;
+        }
+
+    }
+
+    public static String getValidFileName(String fileName) throws IllegalArgumentException{
+
+        if (!isSet(fileName)) {
+
+        	throw new IllegalArgumentException("Invalid Filename passed in: " + fileName);
 
+        } else {
+    	    StringBuffer buffer = new StringBuffer();
```

Comment:
Is there any reason to use StringBuffer instead of StringBuilder, the code looks very thread-safe to use syncro implementation of the CharSequence.